tests/python/pants_test/util/test_objects.py

```diff
@@ -457,26 +477,44 @@ def test_type_check_errors(self):
     # type checking failure with multiple arguments (one is correct)
     with self.assertRaises(TypeCheckError) as cm:
       AnotherTypedDatatype(str('correct'), str('should be list'))
-    expected_msg = (
-      """error: in constructor of type AnotherTypedDatatype: type check error:
-field 'elements' was invalid: value 'should be list' (with type 'str') must satisfy this type constraint: Exactly(list).""")
-    self.assertEqual(str(cm.exception), expected_msg)
+    def compare_string_rep(string_type):
```

Comment:
Perhaps rather than making all of these conditional, you could match against regexes instead? Alternatively, maybe have your helper function do something like:
```
self.assertIn(str(cm).exception, [expected_message_with_str, expected_message_with_newstr])
```
?

Reply:
I originally started that way but changed to these conditionals to align code more with the feature checking idiom. It makes it more explicit this is for the sake of compatibility, and allows us to remove python 2 more easily once we stop supporting it. 

Alas it’s not the prettiest code..

Reply:
Mm, yea... that's reasonable I guess.